optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/inverserelation/InverseRelationShadowVariableDescriptor.java

```diff
@@ -93,16 +96,19 @@ private void linkShadowSources(DescriptorPolicy descriptorPolicy) {
                     + sourceEntityDescriptor.getEntityClass() + ").\n"
                     + entityDescriptor.buildInvalidVariableNameExceptionMessage(sourceVariableName));
         }
-        boolean chained = (sourceVariableDescriptor instanceof GenuineVariableDescriptor) &&
+        chained = (sourceVariableDescriptor instanceof GenuineVariableDescriptor) &&
                 ((GenuineVariableDescriptor<Solution_>) sourceVariableDescriptor).isChained();
+        boolean list = (sourceVariableDescriptor instanceof GenuineVariableDescriptor) &&
+                ((GenuineVariableDescriptor<Solution_>) sourceVariableDescriptor).isListVariable();
         if (singleton) {
-            if (!chained) {
+            if (!chained && !list) {
                 throw new IllegalArgumentException("The entityClass (" + entityDescriptor.getEntityClass()
                         + ") has a @" + InverseRelationShadowVariable.class.getSimpleName()
                         + " annotated property (" + variableMemberAccessor.getName()
                         + ") which does not return a " + Collection.class.getSimpleName()
                         + " with sourceVariableName (" + sourceVariableName
-                        + ") which is not chained. Only a chained variable supports a singleton inverse.");
+                        + ") which is neither chained nor list variable."
+                        + " Only a chained and list variables support a singleton inverse.");
```

Comment:
list first :)

Reply:
"list variables" - do users realize its @PlanningCollectionVariable? 
=> use "...@" + PlanningCollectionVariable.class.getSimpleName() + ... in error.

Reply:
Not sure. We could apply the same logic to chained variables:

> "chained variable" - do users realize it's `@PlanningVariable(graphType = CHAINED`?

I think a "list variable" is good term that makes it immediately clear what kind of variable it is. At least for now, before other collection variable types are introduced.

On the other hand there is currently a discrepancy between the term "list variable" and the API - `@PlanningCollectionVariable`. Why not `@PlanningListVariable`? Why not `@PlanningVarible(graphType = DISJOINT_LISTS)`.

I think this is one of the thinks that can be done as the last step, when the implementation is done, and that need a wider consensus, perhaps a community feedback. And perhaps a bake time before it becomes a public API.

Reply:
we ended up with a compromise in the meeting, right?

Reply:
Yes. The diff is outdated. The compromise has been implemented. Resolving.